iotdb/src/main/java/org/apache/iotdb/db/utils/datastructure/ListPublicBAOSPool.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.utils.datastructure;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Just copy from {@linkplain org.apache.iotdb.db.rescon.PrimitiveArrayPool PrimitiveArrayPool}.
+ *
+ * Provide and recycle {@code byte[]} in {@linkplain ListPublicBAOS}.
+ *
+ * @author Pengze Lv, kangrong
+ */
+public class ListPublicBAOSPool {
```

Comment:
change class name to ByteArrayPool or reuse PrimitiveArrayPool directly (add a byte[] type in PrimitiveArrayPool)

Reply:
It's our initial thought but as advised by @jixuan1989, we separate `ByteArrayPool` from `PrimitiveArrayPool`.